built_value_generator/lib/src/serializer_source_library.dart

```diff
@@ -22,20 +22,33 @@ abstract class SerializerSourceLibrary
       new _$SerializerSourceLibrary._(element: element);
   SerializerSourceLibrary._();
 
-  // TODO(davidmorgan): better way of checking for top level declaration.
   @memoized
-  bool get hasSerializers => element.definingCompilationUnit.accessors
-      .any((element) => element.displayName == 'serializers');
+  bool get hasSerializers => serializersForAnnotation != null;
+
+  /// Returns the @serializersFor annotation from an accessor called
+  /// "serializers", or null if none is found.
+  @memoized
+  ElementAnnotation get serializersForAnnotation {
+    final accessors = element.definingCompilationUnit.accessors
+        .where((element) =>
+            element.isGetter && element.displayName == 'serializers')
+        .toList();
```

Comment:
Unnecessary .toList

Reply:
Without the "toList" we iterate all the accessors twice -- once for "isEmpty" and once for "single". So I think it's nicer to call "toList" here.

Reply:
Technically no, iterable isEmpty is implemented by checking if you can move an iterator, so O(1) as well.
https://api.dartlang.org/stable/1.24.2/dart-core/Iterable/isEmpty.html

But it is just a minor nit.

Reply:
There "where" means that to answer "isEmpty" it has to find the first match -- or iterate over all elements if there aren't any.